.github/workflows/ci-test-publish.yml

```diff
@@ -0,0 +1,47 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different
+# versions of node. Also, if its on master (not a PR) and there is a new version then it will publish the package.
+# See: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: CI Build
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  test:
+    name: Test on Ubuntu node ${{ matrix.node_version }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node_version: [16.x]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node_version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node_version }}
+      - name: Upgrade npm
+        if: matrix.node_version < 16
+        run: npm install -g npm@7
+      - run: npm ci
+      - run: npm run prettier:ci
+      - run: npm run build
+      - run: npm run test:ci
+  publish:
+    name: npm Publish
+    if: github.event_name != 'pull_request'
```

Comment:
Will this publish a new package every time I push to master? On Appveyor, it was setup so that it would only publish when a new tag was created. I would prefer to keep it that way. In another repo, I used `if: startsWith(github.ref, 'refs/tags/v')` on the publish step to get the desired behavior. You also need to make sure the workflow is triggered when a tag is pushed.

Reply:
I believe this will do what we want.